stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -1574,10 +1574,10 @@ private static int getTimeZoneOffset(String timezoneStr) {
         return columnDefs;
     }
 
-    public static Connection getDatabaseConnection(Context context, SQLDatasource datasource, boolean isInTransaction)
-            throws SQLException {
+    public static Connection getDatabaseConnection(Context context, SQLDatasource datasource, boolean isInTransaction,
+            boolean isSelectQuery) throws SQLException {
```

Comment:
Isn't it possible to explicitly set value for "isTransaction"  instead of introducing another param ?